clash-lib/src/Clash/Normalize.hs

```diff
@@ -289,9 +287,9 @@ flattenNode c@(CLeaf (nm,(_,_,_,e))) = do
                return (Right ((nm,mkApps (mkTicks fun ticks) (reverse remainder)),[]))
           _ -> return (Right ((nm,e),[]))
       _ -> return (Right ((nm,e),[]))
-flattenNode b@(CBranch (_,(_,_,NoInline,_)) _) =
+flattenNode b@(CBranch (_,(Binding _ _ NoInline _)) _) =
```

Comment:
I'd prefer to use `Binding{bindingSpec=NoInline}` in these cases, but we don't have to do it for this PR.

Reply:
Yeah, the changes to make things compile again were done fairly mechanically. Some of the use sites are already pretty hairy as-is